src/Kestrel.Core/Internal/Http2/Http2FrameWriter.cs

```diff
@@ -193,6 +193,48 @@ public void WriteResponseHeaders(int streamId, int statusCode, IHeaderDictionary
             }
         }
 
+        public Task WriteResponseTrailers(int streamId, HttpResponseTrailers headers)
+        {
+            lock (_writeLock)
+            {
+                if (_completed)
+                {
+                    return Task.CompletedTask;
+                }
+
+                _outgoingFrame.PrepareHeaders(Http2HeadersFrameFlags.END_STREAM, streamId);
```

Comment:
It looks to me that the bulk of this method could be shared between this and WriteResponseHeaders.

Reply:
Consolidated